Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  MineField.swift
+//  Minesweeper
+//
+//  Created by Dorian Grolaux on 19/02/2018.
+//
+
+import Foundation
+
+typealias Coordinates = (x: Int, y: Int)
+
+class MineField {
+    private let cells: [[Cell]]
+    
+    /**
+     Initialise the Minefield with an already filled field (use for saved games for instance).
+     - parameter cells: The cell 2d array.
+     **/
+    init(cells: [[Cell]]) {
+        self.cells = cells
+    }
+    
+    
+    /**
+     Initialise the Minefield of given square size and randomly placing the given amount of bombs.
+     - parameter size: The length of each side of the field measured in amount of cells. (Total number of cells is amount squared).
+     - parameter bombs: The amount of bombs to randomly place around in the Minefield.
+     **/
+    init(squareSideLength size: Int, withBombAmount bombs: Int) {
+        
+        let randomPlacements = MineField.getRandomMinesPlacements(mineAmount: bombs, forMinefieldTotalCellAmount: size * size)
+        var cells = [[Cell]]()
+        
+        for x in 0..<size {
+            var yCells = [Cell]()
+            
+            for y in 0..<size {
+                yCells.append(Cell(containsBomb: randomPlacements.contains(where: { placement -> Bool in
+                    return placement == (x: x, y: y)
+                })))
+            }
+            cells.append(yCells)
+        }
+        
+        MineField.markPlacedBombs(inField: &cells, placedBombs: randomPlacements)
+        
+        self.cells = cells
+    }
+    
+    
+    /**
+     Retrieve a cell's immediate neighbours (up, down, left, right & diagonally).
+     - parameter cellCoords: The X & Y position of the cell to retrieve neighbours from.
+     - parameter field: The Minefield to look through.
+     - parameter returns: All the coordinates of the given cell's neighbours (up to 8).
+     **/
+    private static func getImmediateNeighbours(ofCellAt cellCoords: Coordinates, inField field: [[Cell]]) -> [Coordinates] {
+        var neighbouringCellsCoords = [Coordinates]()
+        
+        for xDelta in -1...1 {
```

Comment:
This is a bit hard to read.
What is xDelta?
what is x?
What are cellCoords...well ok I can figure that one out but ew.